### PR TITLE
Tweaks and fixes (4)

### PR DESCRIPTION
**Pull request by hcarneiro**
**NOTE:** This needs (https://github.com/Fliplet/fliplet-widget-interactive-map/pull/13) merged before reviewing

- Refactor how users manage marker styles and save them in the background
  - https://cl.ly/0385d6fafc59/Screenshot%202019-02-20%20at%2013.10.54.png
  - https://cl.ly/689ba7cc0a7b/Screenshot%202019-02-20%20at%2013.11.13.png
  - https://cl.ly/144240ed13b4/Screenshot%202019-02-20%20at%2013.11.30.png
- Updated marker information rows - https://cl.ly/aa52d2ba3828/Screenshot%202019-02-20%20at%2016.48.09.png

### CODE REVIEW

File: widget.json [hunk 1]
```diff
@@ -21,12 +21,17 @@
       "vendor/bootstrap-colorpicker.css",
 
       "js/interface.templates.js",
+      "dist/core.js.map",
```
Comment (squallstar) [inline widget.json]: why do you need map files listed? they won't make any different to the backend

Comment (hcarneiro) [reply widget.json]: @squallstar I was having file not found error on iOS because of it

Comment (squallstar) [reply widget.json]: @hcarneiro then just disable source maps from being generated in your gulpfile.

The dist files should not include this line: https://github.com/Fliplet/fliplet-widget-interactive-map/blob/master/dist/add-markers.js#L1567 

That's what is telling iOS to download the map file

**Comment by hcarneiro** [comment]
@squallstar 
Updated PR with:
- Removed and prevented creation source map files
- Improved marker rendering performance - Needs this [Fliplet.UI.PanZoom](https://github.com/Fliplet/fliplet-api/pull/3173) change

File: dist/core.js [hunk 2]
```diff
@@ -93,36 +93,7 @@
...
-        throw new Error('A template for the ' + componentName + ' component has not been found');
-      }
-
-      _component.template = template();
-      Vue.component(componentName, _component);
-    }
-  };
-}();
+eval("Fliplet.InteractiveMap = function () {\n  var components = {};\n  var eventHub = new Vue();\n  var templates = Fliplet.Widget.Templates;\n  return {\n    on: function on(eventName, fn) {\n      eventHub.$on(eventName, fn);\n    },\n    off: function off(eventName, fn) {\n      eventHub.$off(eventName, fn);\n    },\n    emit: function emit(eventName, data) {\n      eventHub.$emit(eventName, data);\n    },\n    component: function component(componentName, _component) {\n      if (!componentName || !_component.componentName) {\n        throw new Error('The component name is required');\n      }\n\n      var template = templates['templates.interface.' + componentName];\n\n      if (!template) {\n        throw new Error('A template for the ' + componentName + ' component has not been found');\n      }\n\n      _component.template = template();\n      Vue.component(componentName, _component);\n    }\n  };\n}();\n\n//# sourceURL=webpack:///./js/libs/core.js?");
```
Comment (squallstar) [inline dist/core.js]: how come it's inside an eval now?

Comment (hcarneiro) [reply dist/core.js]: @squallstar not sure, as you can see I only changed one line in the gulpfile

**Pull request merged**